File: Jump Game/code.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let n = nums.length;

    let index = 0;
    for(let i=0; i<n; i++){
        if(i > index) return false;
        index = Math.max(index, i+nums[i]);
    }
    return true;
};

File: Jump Game/code.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        t = len(nums)-1
        for i in range(len(nums)-2, -1, -1):
            if i+nums[i] >= t:
                t = i
        if t == 0:
            return True
        return False
